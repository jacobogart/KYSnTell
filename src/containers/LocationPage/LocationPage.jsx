import React, { Component } from 'react';
import { connect } from 'react-redux';
import Map from '../../components/Map/Map';

class LocationPage extends Component {
  render() {
    const id = this.props.location.pathname.split('/')[3];
    const { title, address, point, link, telephone } = this.props.locations.find(location => location.id === id);

    return (
      <article className='LocationPage'>
        <Map point={point}/>
        <div className="location-info">
          <h3>{title}</h3>
          <p>{address.split(';')[0]} <br /> {address.split(';')[1]}</p>
          <div className="link-holder">
            <a href={`tel:${telephone}`}><i class="fas fa-phone"></i>{telephone}</a>
            <a href={link}><i className="fas fa-globe"></i>Website</a>
          </div>
        </div>
      </article>
    );
  }
}

export const mapStateToProps = (state) => ({
  locations: state.locations
})

export default connect(
  mapStateToProps,
)(LocationPage);