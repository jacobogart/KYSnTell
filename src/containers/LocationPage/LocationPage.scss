.LocationPage {
  margin-top: 55px;
  height: 70vh;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  .location-info {
    background-color: grey;
    padding: 10px;
    font-size: 6vw;
    width: 80%;
    height: 27vh;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;

    h3 {
      margin: 0;
    }

    p {
      font-size: 6vw;
      margin: 10px 0;
    }

    a {
      color: white;
    }

    .fas {
      margin-right: 5px;
    }

    .link-holder {
      display: flex;
      width: 85%;
      justify-content: space-between;
    }
  }
}