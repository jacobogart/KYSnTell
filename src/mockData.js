import { locationCleaner } from "./api/cleaners";

export const origin = {
  lat: 38.9119463, 
  lng: -94.4136213 
}

export const destination = { 
  lat: "39.1952922", 
  lng: "-94.6817269" 
}

export const distance = '18.3 mi';

export const distanceResult = {
  rows: [
    {
      elements: [
        {
          distance: {
            text: distance
          }
        }
      ]
    }
  ]
}


export const fetchDistanceURL = "https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/distancematrix/json?origins=38.9119463,-94.4136213&destinations=39.1952922,-94.6817269&units=imperial&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const latLongResult = {
  results: [
    {
      geometry: {
        location: origin
      }
    }
  ]
}


export const fetchLatLongURL = "https://maps.googleapis.com/maps/api/geocode/json?address=64081&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const locations = [
  {
    title: "Neighborhood Health",
    link: "https://gettested.cdc.gov/gettested_redirect/21952",
    id: "tag:hivtest.org,2000-03-20:/search/Detailx.asp?id=21952",
    streetAddress: "617 S 8th St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206-3819",
    telephone: "(615)-227-3000",
    categories: "",
    point: {
      lat: "36.1678930",
      long: "-86.7547047"
    },
    updated: "2019-02-08T17:19:32.0000000-05:00"
  },
  {
    title: "Street Works",
    link: "https://gettested.cdc.gov/gettested_redirect/110792",
    id: "tag:hivtest.org,2008-05-21:/search/Detailx.asp?id=110792",
    streetAddress: "520 Sylvan St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206",
    telephone: "(615)-259-7676",
    categories: "",
    point: {
      lat: "36.1665114",
      long: "-86.7605114"
    },
    updated: "2018-08-02T18:05:30.0000000-04:00"
  }
];

export const locationsResult = {
  services: [
    {
      providers: locations
    }
  ]
}

export const cleanLocations = locations.map(location => locationCleaner(location))

export const fetchLocationsURL = "https://cors-anywhere.herokuapp.com/https://locator.hiv.gov/data?lat=38.9119463&long=-94.4136213&distance=10"

export const contact = '+18168682142';

export const contacts = [contact, contact]

export const message = 'Test message.'

export const sendMessageOptions = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({ to: contact, message })
}

export const resolvedMessages = [{ ok: true }, { ok: true }];

export const cleanLocation = {
  title: 'Neighborhood Health',
  link: 'https://gettested.cdc.gov/gettested_redirect/21952',
  id: '21952',
  address: '617 S 8th St;Nashville, TN 37206-3819',
  telephone: '(615)-227-3000',
  point: { lat: '36.1678930', lng: '-86.7547047' }
}

export const details = {
  additionalNotes: "Test",
  diagnosis: "Herpes",
  timeFrame: "year"
}

export const user = {
  location: origin
}

export const searchChangeEvent = {
  target: {
    name: 'zipcode',
    value: '64081'
  }
}

export const submitEvent = {
  preventDefault: jest.fn()
}

export const history = {
  push: jest.fn()
}

export const setUserLocation = jest.fn();

export const setLocations = jest.fn();

export const setContacts = jest.fn();

export const fullMessage = 'Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider. Additional notes from partner: Test'

export const messageWONotes = "Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider."

export const previewEvent = {
  target: {
    blur: jest.fn()
  }
}

export const messageContext = {
  focus: jest.fn(),
  blur: jest.fn(),
  innerText: 'Test'
}

 export const rejectedMessages = [{ ok: false, contact }]

 export const routerLocation = {
  pathname:'/kys/locations/21952'
 }

export const detailsChangeEvent = {
  target: {
    name: 'diagnosis',
    value: 'Herpes'
  }
}

export const setDetails = jest.fn();

export const contactChangeEvent = {
  target: {
    value: contact
  }
}

export const contactBadChangeEvent = {
  target: {
    value: 'test'
  }
}

export const searchDefaultState = {
  zipcode: '',
  distance: '',
  isLoading: false,
  error: false
}