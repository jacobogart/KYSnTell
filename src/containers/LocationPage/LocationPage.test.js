import React from 'react';
import { shallow } from 'enzyme';
import { LocationPage } from './LocationPage';
import * as mock from '../../mockData'

describe('LocationPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LocationPage />);
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const defaultState = { zipcode: '', distance: '' }
    expect(wrapper.state()).toEqual(defaultState);
  });
});