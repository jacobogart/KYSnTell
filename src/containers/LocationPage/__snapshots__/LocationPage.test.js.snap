// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPage should match snapshot 1`] = `
<article
  className="LocationPage"
>
  <div
    className="location-info"
  >
    <h3>
      Neighborhood Health
    </h3>
    <p>
      617 S 8th St
       
      <br />
       
      Nashville, TN 37206-3819
    </p>
    <div
      className="link-holder"
    >
      <a
        href="tel:(615)-227-3000"
      >
        <i
          className="fas fa-phone"
        />
        (615)-227-3000
      </a>
      <a
        href="https://gettested.cdc.gov/gettested_redirect/21952"
      >
        <i
          className="fas fa-globe"
        />
        Website
      </a>
    </div>
  </div>
  <Map
    point={
      Object {
        "lat": "36.1678930",
        "lng": "-86.7547047",
      }
    }
  />
</article>
`;
